### PR TITLE
[UXP-3241] Card form component wrapper

### PR DESCRIPTION
**Pull request by andrejak**
This PR adds our card form component, which is basically a wrapper for the iframe. 
It mainly transforms the props given by the customer into parameters to pass to the iframe, and renders it. 

I added a few stories with different things you can customise.

Documentation: https://www.notion.so/duffel/Card-passthrough-component-cd454b43bde444cea3d44a63ced92f92?pvs=4

Screenshot 2023-11-24 at 14 11 17

If you want to develop with this locally (eg using Storybook), you're going to need to comment out the x-frame-options header and CSP headers that are added to the iframe in `set_security_headers.ex` file, since they don't allow requests from localhost.

### CODE REVIEW

File: src/components/DuffelCardForm/DuffelCardForm.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  styles,
+  submitButtonLabel,
+  onSubmit,
+}) => {
+  React.useEffect(() => {
+    // add event handler to listen to iframe message
+    const iFrameEventListener = (event: MessageEvent) => {
+      if (!TOKEN_PROXY_IFRAME_BASE_URL.startsWith(event.origin)) return;
+      if ("error" in event.data) {
+        // console.error(event.data.error);
```
[the remark this answers is not in the document]
Comment (andrejak) [reply src/components/DuffelCardForm/DuffelCardForm.tsx]: Yeah I'll remove it for now. It made me wonder if we should provide some kind of onError handler, but I think the customer can just listen for the error message and do it that way...

File: src/components/DuffelCardForm/DuffelCardForm.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  clientKey: string;
+
+  styles?: CustomStyles & { stylesheetUrl?: string };
+
+  submitButtonLabel?: string;
+  onSubmit?: (data: TokenisedCardData) => void;
+}
+
+// const TOKEN_PROXY_IFRAME_BASE_URL = "https:localhost:8000/iframe.html"; /// For local development and testing
+const TOKEN_PROXY_IFRAME_BASE_URL = "https://api.duffel.cards/iframe.html";
```
Comment (igorp1) [inline src/components/DuffelCardForm/DuffelCardForm.tsx:14]: wonder if we can add this to the .env files. your call, might be more helpful (or not!)

File: src/stories/DuffelCardForm.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    styles: { accentColor: "29, 78, 216" },
+  },
+};
+
+export const WithFont: DuffelCardFormStory = {
+  args: {
+    ...defaultProps,
+    styles: {
+      fontFamily: `'Tangerine', serif`,
+      stylesheetUrl: "https://fonts.googleapis.com/css?family=Tangerine",
```
Comment (igorp1) [inline src/stories/DuffelCardForm.stories.tsx:34]: How is the stylesheet applied?

File: src/components/DuffelCardForm/DuffelCardForm.tsx (hunk 2 above)
Comment (andrejak) [reply src/components/DuffelCardForm/DuffelCardForm.tsx:14]: Yeah that is probably the right place to do it, will move it.

File: scripts/build-and-publish.sh [hunk 4]
```diff
@@ -48,7 +50,7 @@ if [[ "$@" != *"--dry-run"* ]]; then
     
     # Publish to npm
     cd react-dist
-    npm publish
+    npm publish --canary
```
Comment (igorp1) [inline scripts/build-and-publish.sh:53]: remove this

**Review by igorp1 — APPROVED** [review]

**Pull request merged**